### PR TITLE
#167165006 Role Based Access Control

### PR DESCRIPTION
**Pull request by allebd**
#### Pivotal tracker story
[#167165006](https://www.pivotaltracker.com/story/show/167165006)
#### What does this PR do?
Control user access based on role
#### Summary of Task
- Create role model
- Create function to control users access
- Add test
- Add swagger documentation
#### How can this be tested?
- Use **git clone** to clone this repository to local branch
- Use **npm install** to download all necessary packages
- Ensure your database is created or you can use **sequelize db:create** if database does not exist
- Use **npm run test**  to test
- Go your POSTMAN and try to access a route with a different access priviledge
#### Screenshots (if appropriate)
Role based access control
Flow Chart of time to read a novel

Screenshot 2019-08-15 at 4 30 15 AM
Screenshot of an unaccessible novel page by reader

### CODE REVIEW

File: src/middlewares/authorizeUser.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import helpers from '../helpers';
+
+const { responseMessage } = helpers;
+/**
+ *
+ * @param {object} roleArray an array of roles
+ * @returns {*} json or next
+ */
+const authorizeUser = roleArray => async (request, response, next) => {
```
Comment (WilliamsOhworuka) [inline src/middlewares/authorizeUser.js:9]: Nice one, never thought of using currying in situations like this. I've definitely learned something. keep up the good work.

**Review by WilliamsOhworuka — APPROVED** [review]
LGTM

**Review by OvieMudi — APPROVED** [review]
LGTM

**Review by Rythae — APPROVED** [review]

File: src/middlewares/authorizeUser.js (hunk 1 above)
Comment (allebd) [reply src/middlewares/authorizeUser.js:9]: Thank you for the commendation, I appreciate your feedback

**Pull request merged**